src/overlays/actors/ovl_En_Floormas/z_en_floormas.c

```diff
@@ -9,123 +15,1120 @@ void EnFloormas_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    this->collider.dim.height = (cylinderInit.dim.height * 100.0f) * this->actor.scale.x;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A18500.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->actor.scale.x >= 0.01f) {
+            this->actor.flags &= ~0x10;
+            EnFloormas_Vulnerable(this);
+            this->actor.params = 0;
+            this->collider.body.bumperFlags |= 4;
+            this->actor.colChkInfo.health = colCheckInfoInit.health;
+            EnFloormas_StandBegin(this);
+        } else {
+            if (this->actionTimer == 0) {
+                SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_060039B0);
+                this->actionTimer = 1;
+            } else {
+                SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06009DB0);
+                this->actionTimer = 0;
+            }
+        }
+    }
+    func_8002F974(&this->actor, 0x3133);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A185C4.s")
+void EnFloormas_SmWait(EnFloormas* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A18650.s")
+void EnFloormas_TakeDamage(EnFloormas* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            EnFloormas_Die(this, globalCtx);
+        } else {
+            EnFloormas_RecoverBegin(this);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A187AC.s")
+    if (func_800A56C8(&this->skelAnime, 13.0f)) {
+        if (this->actor.scale.x > 0.004f) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+        }
+    }
+    Math_ApproxF(&this->actor.speedXZ, 0.0f, 0.2f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A187E4.s")
+void EnFloormas_Recover(EnFloormas* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        EnFloormas_StandBegin(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A188F0.s")
+void EnFloormas_Freeze(EnFloormas* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    if (this->actionTimer == 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            EnFloormas_Die(this, globalCtx);
+            return;
+        }
+        EnFloormas_StandBegin(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Floormas/func_80A18AB4.s")
+// EnFloormas_ColliderCheck
+void EnFloormas_ColliderCheck(EnFloormas* this, GlobalContext* globalCtx) {
+    char pad[4];
```

Comment:
we generally make pads s32 not char though its not that big of a deal. If you end up changing it, its in a few places

Reply:
also name comment